### PR TITLE
Adding support for individual marketplace tags

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4344)

## Description
Successful upload flow: https://code.pan.run/xsoar/content/-/pipelines/3996142

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -2346,7 +2346,20 @@ def _collect_pack_tags(self, user_metadata, landing_page_sections, trending_pack
...
         return tags
 
-    def _enhance_pack_attributes(self, index_folder_path, dependencies_metadata_dict,
+    def _get_tags_by_marketplace(self, user_metadata: dict, marketplace: str):
+        """ Returns tags in according to the current marketplace"""
+        tags = []
+        for tag in user_metadata.get('tags', []):
+            if ':' in tag:
+                tag_data = tag.split(':')
+                if tag_data[0] == marketplace:
```
Comment (yaakovpraisler) [inline Tests/Marketplace/marketplace_services.py:2355]: marketplace here can be `xsoar` or `marketplacev2`, not `xsiam`.

File: Packs/AWS-GuardDuty/pack_metadata.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
         "Cloud Services"
     ],
     "tags": [
-        "Data Source"
+        "xsiam:Data Source"
```
Comment (tkatzir) [inline Packs/AWS-GuardDuty/pack_metadata.json:14]: The prefix is `xsiam`? No `marketplacev2`?

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 3]
```diff
@@ -255,9 +255,26 @@ def test_section_tags_added(self, dummy_pack_metadata, dummy_pack):
...
         }
-        tags = dummy_pack._collect_pack_tags(dummy_pack_metadata, section_tags, [])
+        tags = dummy_pack._collect_pack_tags(dummy_pack_metadata, section_tags, [], 'xsoar')
         assert 'Featured' in tags
 
+    @pytest.mark.parametrize('pack_metadata,marketplace,expected_result',
+                             [({'tags': ['tag1', 'marketplacev2:tag2']}, 'xsoar', {'tag1'}),
+                              ({'tags': ['tag1', 'marketplacev2:tag2']}, 'marketplacev2', {'tag1', 'tag2'}),
+                              ({'tags': ['marketplacev2:tag2']}, 'xsoar', set()),
+                              ({'tags': ['tag1', 'marketplacev2,xsoar:tag2']}, 'xsoar', {'tag1', 'tag2'})])
```
Comment (tkatzir) [inline Tests/Marketplace/Tests/marketplace_services_test.py:265]: Add a test for non-existent marketplace.

File: Tests/Marketplace/marketplace_services.py [hunk 4]
```diff
@@ -2346,7 +2346,20 @@ def _collect_pack_tags(self, user_metadata, landing_page_sections, trending_pack
...
         return tags
 
-    def _enhance_pack_attributes(self, index_folder_path, dependencies_metadata_dict,
+    def _get_tags_by_marketplace(self, user_metadata: dict, marketplace: str):
+        """ Returns tags in according to the current marketplace"""
+        tags = []
+        for tag in user_metadata.get('tags', []):
+            if ':' in tag:
+                tag_data = tag.split(':')
+                if marketplace in tag_data[0]:
```
Comment (tkatzir) [inline Tests/Marketplace/marketplace_services.py:2355]: Need to split to multiple marketplaces, with a comma as a separator.

**Review by tkatzir — CHANGES_REQUESTED** [review]
Looks good!
Some minor comments, and I'm missing a failure test case, and the support for multiple marketplaces.

File: Packs/AWS-GuardDuty/pack_metadata.json (hunk 2 above)
Comment (MLainer1) [reply Packs/AWS-GuardDuty/pack_metadata.json:14]: Yes, will change

File: Tests/Marketplace/marketplace_services.py (hunk 4 above)
Comment (MLainer1) [reply Tests/Marketplace/marketplace_services.py:2355]: Actually if we got to this point, the marketplaces are valid (There will be a validation in the demisto-sdk for it)

File: Tests/Marketplace/Tests/marketplace_services_test.py (hunk 3 above)
Comment (MLainer1) [reply Tests/Marketplace/Tests/marketplace_services_test.py:265]: Actually if we got to this point, the marketplaces are always valid (There will be a validation in the demisto-sdk for it)

File: Packs/AWS-GuardDuty/pack_metadata.json (hunk 2 above)
Comment (yaakovpraisler) [reply Packs/AWS-GuardDuty/pack_metadata.json]: @tkatzir - I think it should be `xsiam` and not `marketplacev2` as we are planning in the future to not use marketplacev2 anymore and use xsiam instead.

File: Tests/Marketplace/marketplace_services.py (hunk 4 above)
Comment (dantavori) [reply Tests/Marketplace/marketplace_services.py:2355]: I agree with Tal here, it's a small change to add support for this

**Review by dantavori — COMMENTED** [review]
looks good, of course - should be merged only when we release the SDK version with your changes

File: Tests/Marketplace/marketplace_services.py (hunk 4 above)
Comment (MLainer1) [reply Tests/Marketplace/marketplace_services.py:2355]: @dantavori Talked with Tal about it, tag_data[0] is a string that contains all of the marketplaces that are valid for the current tag, so if `marketplace == "xsoar"` and `tag_data[0] == "xsoar,marketplacev2"`, we will include the tag in the current pack upload

**Review by tkatzir — APPROVED** [review]

**Pull request merged**